runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -575,6 +585,76 @@ function M.stop(bufnr, client_id)
   end
 end
 
+--- Return the semantic token(s) at the given position.
+--- If called without argument, returns the token under the cursor.
+---
+---@param bufnr number|nil Buffer number (0 for current buffer, default)
+---@param row number|nil Position row (default cursor position)
+---@param col number|nil Position column (default cursor position)
+---
+---@return table[]|nil tokens Table of tokens at position
+function M.get_at_pos(bufnr, row, col)
+  if bufnr == nil or bufnr == 0 then
+    bufnr = api.nvim_get_current_buf()
+  end
+
+  local highlighter = STHighlighter.active[bufnr]
+  if not highlighter then
+    return
+  end
+
+  if not (row and col) then
+    local cursor = api.nvim_win_get_cursor(0)
+    row = cursor[1] - 1
+    col = cursor[2]
+  end
+
+  local tokens = {}
+  for _, client in pairs(highlighter.client_state) do
+    local highlights = client.current_result.highlights
+    if highlights then
+      local idx = binary_search(highlights, row)
+      for i = idx, #highlights do
+        local token = highlights[i]
+
+        if token.line > row then
+          break
+        end
+
+        if token.start_col <= col and token.end_col > col then
+          tokens[#tokens + 1] = token
+        end
+      end
+    end
+  end
+  return tokens
+end
+
+--- Show the semantic token(s) under the cursor
+function M.show_at_cursor()
+  local tokens = M.get_at_pos()
+  if not tokens then
+    return
+  end
+
+  local msg = {}
+  for _, token in pairs(tokens) do
+    local type = '@' .. token.type
+    msg[#msg + 1] = { 'type: ' }
+    msg[#msg + 1] = { type, type }
+    if #token.modifiers > 0 then
+      msg[#msg + 1] = { '  modifiers:' }
+      for _, modifier in pairs(token.modifiers) do
+        local mod = '@' .. modifier
+        msg[#msg + 1] = { ' ' }
+        msg[#msg + 1] = { mod, mod }
+      end
+    end
+    msg[#msg + 1] = { '    ' }
+  end
+  api.nvim_echo(msg, false, {})
```

Comment:
what is the difference between "show", "echo", "print", and for some reason "vim.notify"?

Reply:
`vim.notify` is an interface which plugins can use to implement fancy ui's for notifications. See https://github.com/rcarriga/nvim-notify.

Not sure myself on the difference between `show`, `echo` and `print`.